spec/queries/subscriber_list_query_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'rails_helper'
+
+RSpec.describe SubscriberListQuery do
+  subject do
+    described_class.new(
+      tags: { policies: ['eggs'] },
+      links: { policies: ['11aa'] },
+      document_type: 'travel_advice',
+      email_document_supertype: 'publications',
+      government_document_supertype: 'news_stories'
+    )
+  end
+
+  shared_examples '#links matching' do |tags_or_links|
+    it { is_included_in_links tags_or_links, document_type: 'travel_advice' }
+    it { is_excluded_from_links tags_or_links, document_type: 'other' }
+    it { is_included_in_links tags_or_links, email_document_supertype: 'publications' }
+    it { is_excluded_from_links tags_or_links, email_document_supertype: 'other' }
+    it { is_included_in_links tags_or_links, government_document_supertype: 'news_stories' }
+    it { is_excluded_from_links tags_or_links, government_document_supertype: 'other' }
+
+    it do
+      is_included_in_links(
+        tags_or_links,
+        document_type: 'travel_advice',
+        email_document_supertype: 'publications',
+        government_document_supertype: 'news_stories',
+      )
+    end
+
+    it do
+      is_excluded_from_links(
+        tags_or_links,
+        document_type: 'other',
+        email_document_supertype: 'publications',
+        government_document_supertype: 'news_stories',
+      )
+    end
+
+    it do
+      is_excluded_from_links(
+        tags_or_links,
+        document_type: 'travel_advice',
+        email_document_supertype: 'other',
+        government_document_supertype: 'news_stories',
+      )
+    end
+
+    it do
+      is_excluded_from_links(
+        tags_or_links,
+        document_type: 'travel_advice',
+        email_document_supertype: 'publications',
+        government_document_supertype: 'other',
+      )
+    end
+  end
+
+  context 'when matching has tags fields' do
+    it_behaves_like "#links matching", { tags: { policies: ['eggs'] }, links: {} }
+
+    it "excluded when non-matching tags" do
+      subscriber_list = create_subscriber_list(tags: { policies: ['apples'] }, document_type: 'document_type')
```

Comment:
Actually we don't have any tests yet for this query for the cases where the list has no document/super type - only for ones where it matches or is a different type, rather than being `''`.